### PR TITLE
replace assert

### PR DESCRIPTION
**Pull request by boa85**

### CODE REVIEW

**Comment by atcsecure** [comment]
@corvaxx the assert/crash is fairly common and impacting trading, root cause should be found as why we keep getting "bad role" error

**Comment by corvaxx** [comment]
you right, should be found. before release
this assert shows abnormal situations with generated secret (side mismatch, early disclose of, etc)
at this point user B MUST NOT KNOW secret field

File: src/xbridge/xbridgesession.cpp [hunk 1]
```diff
@@ -1554,7 +1554,10 @@ bool Session::Impl::processTransactionCreate(XBridgePacketPtr packet)
 
     if (xtx->role == 'B')
     {
-        assert(xtx->xPubKey.size() == 0 && "bad role");
+        if(xtx->xPubKey.size() != 0) {
```
Comment (magic53) [inline src/xbridge/xbridgesession.cpp]: @boa85 `if(xtx->xPubKey.size() == 0) {`

**Comment by magic53** [comment]
if roleA and roleB both have same address in wallet, roleA can also receive roleB packet. so returning false should be safe

**Comment by corvaxx** [comment]
fixed in 227, assert can be removed

**Comment by corvaxx** [comment]
Look, assert in this place forced to understand the error situation, otherwise nobody would simply not pay attention to this

**Pull request closed without merging**